### PR TITLE
Change `http.request.body.size` and `http.response.body.size` attributes from recommended to opt-in

### LINKED ISSUE #456: Are `http.(request|response).body.size` worth being recommended on HTTP spans (vs opt-in)?
While these can be useful, they are not as generally useful as the other recommended attributes on HTTP spans.

And [`http.(client|server).(request|response).body.size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpserverrequestbodysize) metrics (and exemplars) seem to cover some of their usefulness.

As we think about telemetry cost, I think it's worth asking if these are worth being recommended vs opt-in.

### PR DESCRIPTION
**Pull request by trask**
Fixes #456

## Changes

There are a few concerns with `http.request.body.size` and `http.response.body.size`:

* They are not useful enough to justify their current "Recommended" requirement level
* They are defined in terms of the `Content-Length` header which can already be captured as `http.request.header.content-length` and `http.response.header.content-length`
* They should potentially be defined in terms of actual bytes read, since `Content-Length` is not always available, and actual bytes read could be different anyways if the request/response was cancelled.

This PR does two things:
* Removes them from HTTP semconv (effectively making them opt-in)
* Explicitly marks them as `experimental`, along with a comment not to mark them stable when we mark the other HTTP attributes as stable next week

I will send a separate PR to propose changes to the definition of these attributes based on the other concerns above.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* ~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]

File: model/registry/deprecated/http.yaml [hunk 1]
```diff
@@ -31,11 +31,11 @@ groups:
         examples: ['/search?q=OpenTelemetry#SemConv']
       - id: request_content_length
         type: int
-        brief: 'Deprecated, use `http.request.body.size` instead.'
+        brief: 'Deprecated, use the opt-in `http.request.header.content-length` instead.'
```
Comment (AlexanderWert) [inline model/registry/deprecated/http.yaml:34]: Here, we are in the registry where the requirement level is sort of meaningless. Why do we need that `... the opt-in ...` addition here?

File: model/registry/deprecated/http.yaml [hunk 2]
```diff
@@ -31,11 +31,11 @@ groups:
...
         stability: deprecated
         examples: 3495
       - id: response_content_length
         type: int
-        brief: 'Deprecated, use `http.response.body.size` instead.'
+        brief: 'Deprecated, use the opt-in `http.response.header.content-length` instead.'
```
Comment (AlexanderWert) [inline model/registry/deprecated/http.yaml:39]: same as above

File: model/trace/http.yaml [hunk 3]
```diff
@@ -10,10 +10,8 @@ groups:
       - ref: http.request.method_original
         requirement_level:
           conditionally_required: If and only if it's different than `http.request.method`.
-      - ref: http.request.body.size
```
Comment (AlexanderWert) [inline model/trace/http.yaml:13]: How about keeping the reference here, but marking them explicitly as `opt_in`.
I think there's some value in having them listed here _explicitly_ as opt-in as these attributes are "meaningful" opt-in attributes in this specific context (compared to other _implicit_ opt-in attributes, let's say `rpc.grpc.request.metadata`). So same as we do with `http.request.header` here.

Or is the reason to remove it here the stability level of this entire attributes group being stable while `http.request.body.size` will stay experimental?

Comment (trask) [reply model/trace/http.yaml:13]: > How about keeping the reference here, but marking them explicitly as `opt_in`. I think there's some value in having them listed here _explicitly_ as opt-in as these attributes are "meaningful" opt-in attributes in this specific context (compared to other _implicit_ opt-in attributes, let's say `rpc.grpc.request.metadata`). So same as we do with `http.request.header` here.

good point

do you think we should bring back `network.type` as an explicit opt-in attribute? (#410)

> Or is the reason to remove it here the stability level of this entire attributes group being stable while `http.request.body.size` will stay experimental?

for simplicity, I think I'd prefer to remove it for now for this reason, and after marking HTTP semconv stable we can start figuring out how best to bring in experimental portions, and/or agree on making these particular attributes stable soon (#461)

**Comment by AlexanderWert** [comment]
@trask I think you need to regenrate the tables

**Review by AlexanderWert — APPROVED** [review]

**Comment by pyohannes** [comment]
I guess we should then consider doing the same for messaging ([`messaging.message.body.size` and `messaging.message.envelope.size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#messaging-attributes)) and RPC ([`message.compressed_size` and `message.uncompressed_size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#events))?

**Review by joaopgrassi — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by trask** [comment]
> I guess we should then consider doing the same for messaging ([`messaging.message.body.size` and `messaging.message.envelope.size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#messaging-attributes)) and RPC ([`message.compressed_size` and `message.uncompressed_size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#events))?

I opened #474 to discuss/track

**Pull request merged**